batch/batch/driver/main.py

```diff
@@ -273,13 +273,37 @@ async def deactivate_instance_1(instance):
     log.info(f'deactivating {instance}')
     await instance.deactivate('deactivated')
     await instance.mark_healthy()
-    return web.Response()
 
 
 @routes.post('/api/v1alpha/instances/deactivate')
 @active_instances_only
 async def deactivate_instance(request, instance):  # pylint: disable=unused-argument
-    return await asyncio.shield(deactivate_instance_1(instance))
+    await asyncio.shield(deactivate_instance_1(instance))
+    return web.Response()
+
+
+@routes.post('/instances/{instance_name}/deactivate')
+@check_csrf_token
+@web_authenticated_developers_only()
+async def kill_instance(request, userdata):
+    instance_name = request.match_info['instance_name']
+
+    inst_coll_manager: InstanceCollectionManager = request.app['driver'].inst_coll_manager
+    instance = inst_coll_manager.get_instance(instance_name)
```

Comment:
I'd like to strive for IDE-navigable types in new code that we introduce. If `get_instance` had types in its signature, then the language server will be able to infer the type of `instance` and we could then navigate to the definition of methods like `instance.kill` below. Could you add a type signature to `get_instance`? It should return an optional instance.